### ISSUE TITLE
3.10 windows tests failing

### ISSUE THREAD

**Issue by edublancas** [ploomber/ploomber#830]
CI on windows for python 3.10 is failing: https://github.com/ploomber/ploomber/runs/6708215344?check_suite_focus=true

I see it's because these tests require pygraphviz but that library doesn't work with python 3.10, perhaps we should skip them on 3.10?

**Comment by 94rain** [comment]
Pygraphviz-related tests that can be skipped (with keyword pygraphviz/png)
```
FAILED tests/cli/test_custom.py::test_plot  (pygraphviz/png)
FAILED tests/dag/test_dag.py::test_plot_include_products - ValueError: 'inclu...  (pygraphviz)
FAILED tests/dag/test_dag.py::test_plot_path - ValueError: Error when using d...  (pygraphviz)
FAILED tests/dag/test_dag.py::test_to_graph_prepare_for_graphviz - ImportErro… (pygraphviz)
FAILED tests/dag/test_dag.py::test_to_graph_prepare_for_graphviz_include_products (pygraphviz)
FAILED tests/dag/test_dag.py::test_graphviz_graph_with_clashing_task_str - Im… (pygraphviz)
```

Unknown (TODO):
```
FAILED tests/cli/test_custom.py::test_plot_uses_name_if_any (png, but also monkeypatch?)
FAILED tests/dag/test_dag.py::test_plot_embed - TypeError: 'NoneType' object ... (monkeypatch_plot)
FAILED tests/dag/test_dag.py::test_to_markup[None-html] - ValueError: Error w... (why markup also fails?)
FAILED tests/dag/test_dag.py::test_dag_functions_do_not_fetch_metadata[executor0-plot] (monkeypatch_plot)
```

Looks like monkeypatch_plot and markup are intended not to use pygraphviz, but related tests are still failing. I am looking into this.

**Comment by 94rain** [comment]
In Python 3.10 on Windows, we are only able to use d3 as the plotting backend. 

I see that in the `to_markup` method, `self.plot(output=path_to_plot)` will pass the generated .png file to `plot` method, which is before it is converted to html via`image_bytes2html`.
https://github.com/ploomber/ploomber/blob/b03f3cad6e61d8c1b5277c9ef5331cd344626bc1/src/ploomber/dag/dag.py#L812-L816

In `plot` method, this will raise an exception since the suffix is not `.html` (`image_bytes2html` hasn't happened yet). I don't know it is intended or not.
https://github.com/ploomber/ploomber/blob/b03f3cad6e61d8c1b5277c9ef5331cd344626bc1/src/ploomber/dag/dag.py#L880-L881

**Comment by edublancas** [comment]
@idomic

**Comment by edublancas** [comment]
you're right @94rain. this is what a user reported in https://github.com/ploomber/ploomber/issues/845

essentially, when I deployed the D3 backend, I introduced an error since a user without pygraphviz will not be able to use `to_markup()` (or `ploomber report` - which runs `to_mkarkup()` under the hood).

We should add the `backend=None` to `to_markup()` and use this logic:

https://github.com/ploomber/ploomber/blob/908b61971c4480d1887528aa6a67f6c3357cf8f8/src/ploomber/cli/plot.py#L39

to determine what arguments to pass to `plot()` (so we pass the right extension)

this way markup will work with both backends.

the next thing to test is if the HTML returned when using the D3 backend works in the markdown/HTML report returned my `to_markup`.

essentially what we want is `to_markup` to work well with both backends

**Comment by 94rain** [comment]
Except the `test_task_with_client_and_metaproduct_isnt_outdated_rtrns_waiting_download` `cloud_execution` intermitent tests, I have fixed the 3.10 ci issue but I think it would be better to wait until the `Split unit test CI according to platform` PR is merged before I rebase and create the PR.
